pkgs/development/libraries/libdwarf/0.4.nix

```
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchurl, libelf, zlib }:
+
+stdenv.mkDerivation rec {
+  pname = "libdwarf";
+  version = "0.4.0";
+
+  src = fetchurl {
+    url = "https://www.prevanders.net/libdwarf-${version}.tar.xz";
+    # Upstream displays this hash broken into four parts:
+    sha512 = "30e5c6c1fc95aa28a014007a45199160"
+          + "e1d9ba870b196d6f98e6dd21a349e9bb"
+          + "31bba1bd6817f8ef9a89303bed056218"
+          + "2a7d46fcbb36aedded76c2f1e0052e1e";
```

This is not a sha512 hash but a nar sha512 hash. They can be different hence using the hashes from upstream can lead to errors. That is not the case here but still good to know.

Also I don't think we need to do the dance with line breaks here. Nixpkgs uses TOFU everywhere and comparing the hash afterwards is not a common thing that is done.

```suggestion
    sha512 = "30e5c6c1fc95aa28a014007a45199160e1d9ba870b196d6f98e6dd21a349e9bb31bba1bd6817f8ef9a89303bed0562182a7d46fcbb36aedded76c2f1e0052e1e";
```

---

Updated